README
Dist::Zilla::Role::Tempdir



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Role-Tempdir

1.001004 2017-03-10T12:09:08Z 344f468

1.001003 2017-03-10T12:01:20Z e93ce4e
 - Ship previous devrel changes as stable
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 1.001002, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - test: ↓1

1.001002 2016-05-01T00:22:36Z edbd3ac
 - Added MetaConfig data to consumers.
 - This however requires a TRIAL releases, as this change module requires ::Plugin to already be composed in, but our
   tests and documentation unfortunately instructed users to compose it in *before* that, leading to a breakage.

 [Dependencies::Stats]
 - Dependencies changed since 1.001001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +14 ↑3 -3 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1 (recommends: +2)
 - test: +2 (recommends: +1 ↑1 ↓1)

1.001001 2014-08-15T21:39:24Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - tests augmented.
 - whitespace adjusted.
 - POD reorganised.

 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - test: -4 (recommends: ↑1 -2)

 [Documentation]
 - POD reorganised for consistency.

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - update ReportPrereqs test

1.001000 2014-07-29T09:21:17Z
 [00 Major]
 - ::Dir has new 'keepalive' method to mark/unmark scratch dirs for destruction.
 - ::Dir nas new 'keepalive_fail' method to trigger a croak, keep the temporary directory, and tell the user where to
   find it.
 - ::Dir::run_in()  and thus, ::Role::Tempdir::capture_tempdir(), now pass the Dir object itself as $_[0] for easy
   access.
 - ::Dir now generates tempdirs with the prefix "DZ_R_Tempdir_" for easy cleanup if tempdirs are left around.
 - ::Dir now supports property '_tempdir_owner' which will be normalised and included in the tempdir prefix if
   specified.
 - ::Role::Tempdir::capture_tempdir) now automatically passes the name of the consuming plugin as "_tempdir_owner",
   giving consuming plugins clear tempdir locations when things fail.

1.000001 2014-07-29T06:22:12Z
 [Bugfix]
 - encoded content fetching content was broken, and so, emitted files were all empty! Great!

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - configure: ↓1
 - develop: +3 -3 (suggests: +2 -1)
 - test: +3 ↑1 (recommends: +1)

1.000000 2014-05-16T19:09:54Z
 [00 Major]
 - DZil Encoding support added.
 - Internals overhauled to use newer things.
 - Tooling overhauls
 - Version Scheme overhauls

 [01 Minor backwards incompatible API change]
 - Accessing $item->{name} is now gone entirely, use $item->name instead.
 - $item->{state} may also vanish in the near future, stop using it.

 [02 Dzil Encoding Support]
 - Should now be compatible with DZil 5, and reading files from a tempdir will now default to reading their content as
   "encoded_content" to an InMemory object.

 [03 Path/IO Replaced]
 - Usage of Path::Class and File::Slurp have both been eliminated and replaced by Path::Tiny
 - File::Find::Rule is replaced with Path::Iterator::Rule
 - File::Tempdir usage eliminated in favour of custom + Path::Tiny

 [04 Toolkit moved to EUMM]
 - Toolkit is now EUMM instead of Module::Build
 - Toolkit upgrade dependencies now prefer "latest" via "recommends", instead of "requires"

 [05 Version Scheme change]
 - Mantissa length has shortened to 6
 - Version semantics are now X.YYYZZZ
 - This is the primary motivation for the 1.x on the box.

 [Bugfixes]
 - No longer does any ->isa checking for digest modules. This closes #1

 [Dependencies::Aggregate]
 - 5.010001 : 185
 - 5.018002 : 168
 - 5.019009 : 163

 [Dependencies::Stats]
 - Dependencies changed since 0.01053722, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +44 -2 (recommends: -1, suggests: +1 -1)
 - runtime: +4 ↑1 -4
 - test: +4 ↑1 ↓1 -2 (recommends: +4)

0.01053722 2013-04-08T08:04:30Z
 - Maintenance Release for Module::Build 0.4004

 [Dependencies::Noteworthy]
 - Module::Build 0.3601 → 0.4004
 - Test::More 0.88 → 0.98

 [Dependencies::Stats]
 - Dependencies changed since 0.01053721, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +5 (recommends: ↑1, suggests: ↑1)
 - test: ↑1

 [Documentation]
 - Update Copyright year
 - Add README.mkdn

 [Meta]
 - Bug tracker to github issues

 [Packaging]
 - Utilise Module Build 0.4004 test_requires

0.01053721 2012-02-02T19:54:07Z
 - Maintenance release

 [Dependencies::Stats]
 - Dependencies changed since 0.01053720, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: +3 -1
 - test: -1

 [Internals]
 - Name-spaces declare $AUTHORITY
 - Name-spaces no longer declare $VERSION in BEGIN

 [Packaging]
 - Update LICENSE ( Date, Indent, Address )
 - extra tests moved to xt/
 - GIT url's moved to https://
 - Declares x_authority
 - Versions now done via git

0.01053720 2011-04-05T19:34:41Z
 - Maintenance only release, no really user visible changes.

 [Dependencies::Stats]
 - Dependencies changed since 0.01027622, see misc/*.deps* for details
 - develop: +1 (recommends: +1, suggests: +1)
 - test: +1 -1

 [Packaging]
 - move to @Author::KENTNL
 - Ship with .perltidyrc
 - Rework Changes into CPAN::Changes normalized form.
 - Move perlcriticrc to perlcritic.rc
 - ship with weaver.ini

 [Tests]
 - Drop portability tests
 - Drop hand-coded critic tests and replace with generated.
 - Add cpan-changes tests.

0.01027622 2010-07-23T21:29:14Z
 - Dropped the hardcoded Test::Kwalitee xt-test in favour of the TestKwalitee plugin. ( This will hopefully get this
   module and all dependents out of the heavy 100 )
 - Documentation updated a bit.
 - digest_for is now a public method.

 [Dependencies::Stats]
 - Dependencies changed since 0.01021401, see misc/*.deps* for details
 - build: -9
 - runtime: +2
 - test: +8


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Shell Out and collect the result in a DZ plug-in.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Role-Tempdir",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Role::Tempdir",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Digest::SHA" => "5.47",
    "Dist::Zilla" => "2.0",
    "Dist::Zilla::File::InMemory" => 0,
    "File::chdir" => "0.1002",
    "Moose" => "0.92",
    "Moose::Role" => 0,
    "MooseX::LazyRequire" => 0,
    "Path::Iterator::Rule" => 0,
    "Path::Tiny" => 0,
    "Scalar::Util" => 0,
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla" => "2.0",
    "Dist::Zilla::File::FromCode" => 0,
    "Dist::Zilla::Role::FileInjector" => 0,
    "Dist::Zilla::Role::InstallTool" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Tester" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001004",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Digest::SHA" => "5.47",
  "Dist::Zilla" => "2.0",
  "Dist::Zilla::File::FromCode" => 0,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileInjector" => 0,
  "Dist::Zilla::Role::InstallTool" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Tester" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::chdir" => "0.1002",
  "Moose" => "0.92",
  "Moose::Role" => 0,
  "MooseX::LazyRequire" => 0,
  "Path::Iterator::Rule" => 0,
  "Path::Tiny" => 0,
  "Scalar::Util" => 0,
  "Test::More" => 0,
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    accessors
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}
$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4201"
  },
  "configure_requires" => {
    "Module::Build" => "0.4201"
  },
  "dist_abstract" => "Shell Out and collect the result in a DZ plug-in.",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Role-Tempdir",
  "dist_version" => "0.01053723",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Role::Tempdir",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Digest::SHA" => "5.47",
    "Dist::Zilla" => "2.0",
    "Dist::Zilla::File::InMemory" => 0,
    "File::Tempdir" => "0.02",
    "File::chdir" => "0.1002",
    "Moose" => "0.92",
    "Moose::Role" => 0,
    "Path::Iterator::Rule" => 0,
    "Path::Tiny" => 0,
    "Scalar::Util" => 0,
    "namespace::autoclean" => "0.09",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla" => "2.0",
    "Dist::Zilla::Role::FileInjector" => 0,
    "Dist::Zilla::Role::InstallTool" => 0,
    "Dist::Zilla::Tester" => 0,
    "Test::More" => "1.001002"
  }
);


my %fallback_build_requires = (
  "Dist::Zilla" => "2.0",
  "Dist::Zilla::Role::FileInjector" => 0,
  "Dist::Zilla::Role::InstallTool" => 0,
  "Dist::Zilla::Tester" => 0,
  "Module::Build" => "0.4201",
  "Test::More" => "1.001002"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
